### DISCUSSION TITLE
Can this be automated?

### DISCUSSION THREAD

**Discussion by bigplayer-ai** [aviolaris/instaunfollowers#3] [Q&A] [answered]
Hey,
I have a raspberry-pi server, I was thinking If I can automate this to be done every week, so I get a summary of my Un-followers or maybe a telegram bot that sends me every week who unfollowed me if there are any Un-followers.

**Comment by aviolaris** [comment, marked as answer]
Hello!

The aforementioned automation would require the application to obtain the credentials of each user and use them to send automated requests to Instagram servers at random intervals to check for the availability of the information file for download after it has been requested. However, implementing such functionality would conflict with the primary objective of this application, which is to avoid violating Instagram's terms of use and prevent potential bans. As a result, there are currently no plans to integrate such a feature.

Nevertheless, if you're willing to take the risk of a possible ban, you could try the approach I mentioned earlier. This involves downloading the file once a week, and on successful download, activating the section of the code of the current application that detects unfollowers. This would then provide you with the desired list, which you will be able to send to a specific contact via Telegram. Utilizing libraries like [Requests](https://pypi.org/project/requests/), [BeautifulSoup](https://pypi.org/project/beautifulsoup4/), and [python-telegram-bot](https://pypi.org/project/python-telegram-bot/) can help you save time.
